spec.bs

```diff
@@ -6,57 +6,940 @@ Status: CG-DRAFT
+
+Meanwhile, methods for running {{SharedStorageOperation}}s, along with the {{WindowSharedStorage/worklet}} attribute which is used to call {{Worklet/addModule()}}, are declared in {{WindowSharedStorage}} and exposed to the {{Window}} only, as these are the means by which the {{Window}} interacts with the {{SharedStorageWorklet}}.
+
+On the other hand, methods for getting data from the [=shared storage database=] are declared in {{WorkletSharedStorage}} and exposed to the {{SharedStorageWorklet}} only, in order to carefully control the flow of data read from the [=shared storage database|database=].
+
+<xmp class='idl'>
+  [Exposed=(Window,SharedStorageWorklet)]
+  interface SharedStorage {
+    Promise<any> set(DOMString key,
+                     DOMString value,
+                     optional SharedStorageSetMethodOptions options = {});
+    Promise<any> append(DOMString key,
+                        DOMString value);
+    Promise<any> delete(DOMString key);
+    Promise<any> clear();
+  };
+
+  dictionary SharedStorageSetMethodOptions {
+    boolean ignoreIfPresent = false;
+  };
+</xmp>
+
+  ## The {{WindowSharedStorage}} interface ## {#window}
+
+  The {{WindowSharedStorage}} interface is as follows.
+
+<xmp class='idl'>
+  [Exposed=(Window)]
+  interface WindowSharedStorage : SharedStorage {
+    Promise<any> run(DOMString name,
+                     optional SharedStorageRunOperationMethodOptions options = {});
+    Promise<FencedFrameConfig> selectURL(DOMString name,
                                  FrozenArray<SharedStorageUrlWithMetadata> urls,
                                  optional SharedStorageRunOperationMethodOptions options = {});
 
-  Promise<any> run(DOMString name, optional SharedStorageRunOperationMethodOptions options = {});
+    readonly attribute SharedStorageWorklet worklet;
+  };
+
+  dictionary SharedStorageRunOperationMethodOptions {
+    object data;
+  };
+</xmp>
+
+  ### Window Setter/Deleter Methods ### {#window-setter}
+
+  <div algorithm>
+    The <dfn method for="WindowSharedStorage">set(|key|, |value|, |options|)</dfn> method steps are:
+
+    1. Let |promise| be a new [=promise=].
+    1. Let |context| be {{WindowSharedStorage}}'s {{Window}}'s [=Window/browsing context=].
+    1. [=Assert=] that |context| is not null.
+    1. If |key|'s [=string/length=] exceeds the [=key/maximum length=], reject |promise| with a "{{DataError}}" {{DOMException}}.
+    1. If |value|'s [=string/length=] exceeds the [=value/maximum length=], reject |promise| with a "{{DataError}}" {{DOMException}}.
+    1. If |options| is invalid, reject |promise| with a "{{DataError}}" {{DOMException}}.
+    1. If |options| is empty, create |options|.
```

Comment:
Options cannot be empty since the idl has a default value to an empty struct.  So I don't think you need this step.